### PR TITLE
feat: partially support chrome.tabs.update

### LINKED ISSUE #29091: [Bug]:  PDF containing hyperlinks don't work when clicking on them
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

11.4.6

### What operating system are you using?

Windows

### Operating System Version

Windows 10

### What arch are you using?

x64

### Last Known Working Electron version

_No response_

### Expected Behavior

The link should be opened in the current window.

### Actual Behavior

Nothing happens.

### Testcase Gist URL

https://gist.github.com/80a9108e271ca83c82cc4a8772e371fc

### Additional Information

When the PDF is loaded, try to click on the link (https://www.youtube.com/watch?v=KFPtxCDUPqs&feature=youtu.be)

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
Adds partial support for the `chrome.tabs.update` extensions API, to support
link clicks in PDFs.

Closes #29091.

#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Clicking a hyperlink in a PDF now does what you'd expect.

### CODE REVIEW

File: shell/browser/extensions/api/tabs/tabs_api.cc [hunk 1]
```diff
@@ -319,4 +322,182 @@ ExtensionFunction::ResponseAction TabsSetZoomSettingsFunction::Run() {
...
+  if (!url.is_valid()) {
+    const char kInvalidUrlError[] = "Invalid url: \"*\".";
+    *error = ErrorUtils::FormatErrorMessage(kInvalidUrlError, url_string);
+    return false;
+  }
+
+  // Don't let the extension crash the browser or renderers.
+  if (IsKillURL(url)) {
+    const char kNoCrashBrowserError[] =
+        "I'm sorry. I'm afraid I can't do that.";
```
Comment (VerteDinde) [inline shell/browser/extensions/api/tabs/tabs_api.cc:377]: 😂 I actually like this message, but would it be possible for us to give the user more of a hint about what went wrong? Looking at `kill_hosts[]` above, would a user be trying to intentionally kill their tab/app by passing one of these URLs in?

**Review by VerteDinde — APPROVED** [review]
LGTM! Tested the original issue, which is now fixed, and code looks great. Had one question about an error message, not blocking 👍

File: shell/browser/extensions/api/tabs/tabs_api.cc (hunk 1 above)
Comment (nornagon) [reply shell/browser/extensions/api/tabs/tabs_api.cc:377]: This is actually copy-pasted directly from Chromium: https://source.chromium.org/chromium/chromium/src/+/main:chromecast/browser/extensions/api/tabs/tabs_constants.cc;l=84;drc=479cc17585a64910853e9949b053499ecbeca9a5

This only happens if an extension tries to do something like `chrome.tabs.update(tabId, { url: 'chrome://crash' })`.

**Comment by nornagon** [comment]
CI failures unrelated; merging.

**Comment by richpixel** [comment]
> **Release Notes Persisted**
> 
> > Clicking a hyperlink in a PDF now does what you'd expect.

What is the expected behavior? I'm on electron 14.2.1 and when links are clicked nothing happens. Hovering over the link shows the pointer (as expected). I'm rendering the PDF with an <embed> tag in a Renderer process. Thanks.

**Pull request merged**